api/v1alpha08/sonataflowplatform_services_types.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2023 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha08
+
+// ServicesPlatformSpec describes the desired service configuration for "prod" workflows.
+type ServicesPlatformSpec struct {
+	// Deploys the Data Index service for use by "prod" profile workflows.
+	// +optional
+	DataIndex *ServicesPodTemplateSpec `json:"dataIndex,omitempty"`
+	// Persists service(s) to a datasource of choice. Ephemeral by default.
+	// +optional
+	Persistence *PersistenceOptions `json:"persistence,omitempty"`
```

Comment:
so in the future, when we add the jobs service, we'll have that jobs service and data-index will share the database?
@ricardozanini ^^.

Considering that we won't have to many services, by now, data-index and jobs-service (next step), could it make sense to have a structure like

DataindexServiceSpec {

PodTemplate ServicesPodTemplateSpec,

Persistence PersitenceOptions

}

with this definition users can still share the DB if the want, but we have also the chance to deploy the services with differente databases.

Does it make sense?


}

Reply:
Hi @wmedvede, the idea is to share the same Database for both services, avoiding too complex topologies. Your concern is valid, but I prefer to wait for a demand like this instead of overcomplicating the structure now. In the future, we can have a specific persistence configuration for each service if there's demand for that.